BedrockServer.cpp

```diff
@@ -2020,50 +2001,50 @@ void BedrockServer::_finishPeerCommand(unique_ptr<BedrockCommand>& command) {
 }
 
 void BedrockServer::_acceptSockets() {
-    Socket* s = nullptr;
-    Port* acceptPort = nullptr;
-    while ((s = acceptUnlistedSocket(acceptPort))) {
-        if (SContains(_portPluginMap, acceptPort)) {
-            BedrockPlugin* plugin = _portPluginMap[acceptPort];
-            // Allow the plugin to process this
-            SINFO("Plugin '" << plugin->getName() << "' accepted a socket from '" << s->addr << "'");
-            plugin->onPortAccept(s);
-
-            // Remember that this socket is owned by this plugin.
-            s->data = plugin;
+    // Make a list of ports to accept on.
+    // We'll check the control port, command port, and any plugin ports for new connections.
+    list<reference_wrapper<const unique_ptr<Port>>> portList = {_commandPort, _controlPort};
+    for (auto& p : _portPluginMap) {
+        portList.push_back(reference_wrapper<const unique_ptr<Port>>(p.first));
+    }
+
+    // Try each port.
+    for (auto portWrapper : portList) {
+        const unique_ptr<Port>& port = portWrapper.get();
+        // Skip null ports (if the command or control port are closed).
+        if (!port) {
+            continue;
         }
-    }
-}
 
-STCPManager::Socket* BedrockServer::acceptUnlistedSocket(STCPServer::Port*& portOut) {
-    // Initialize to 0 in case we don't accept anything. Note that this *does* overwrite the passed-in pointer.
-    portOut = 0;
-    Socket* socket = nullptr;
-
-    // See if we can accept on any port
-    lock_guard <decltype(portListMutex)> lock(portListMutex);
-    for (Port& port : portList) {
-        // Try to accept on the port and wrap in a socket
-        sockaddr_in addr;
-        int s = S_accept(port.s, addr, true); // Note that this sets the newly accepted socket to be blocking.
-        if (s > 0) {
-            SDEBUG("Accepting socket from '" << addr << "' on port '" << port.host << "'");
-            socket = new Socket(s, Socket::CONNECTED);
-            socket->addr = addr;
-
-            // Record what port it was accepted on
-            portOut = &port;
-
-            // Start up the thread for this socket.
+        // Accept as many sockets as we can.
+        while (true) {
```

Comment:
We've collapsed `_acceptSockets` and `acceptUnlistedSocket` into just `_acceptSockets` cause it's way simpler. These used to do different things for "listed" versus "unlisted" (i.e., whether they're kept track of in `socketList` or not), but that's gone so we don't need two separate functions now.